sql/catalyst/src/main/scala/org/apache/spark/sql/util/SQLOpenHashSet.scala

```diff
@@ -60,6 +61,57 @@ class SQLOpenHashSet[@specialized(Long, Int, Double, Float) T: ClassTag](
 }
 
 object SQLOpenHashSet {
+  def withNullCheckFunc(
+      dataType: DataType,
+      hashSet: SQLOpenHashSet[Any],
+      handleNotNull: Any => Unit,
+      handleNull: () => Unit): ArrayData => Int => Unit = {
```

Comment:
What's the point of using `ArrayData => Int => Unit`? I think `(ArrayData, Int) => Unit` is simpler and cleaner.

Reply:
Done